recipes/antlr4-cppruntime/all/patches/0001-update-cmakelist.patch

```diff
@@ -0,0 +1,25 @@
+--- CMakeLists.txt
++++ CMakeLists.txt
+@@ -40,8 +40,7 @@
+ endif()
+ 
+ if(CMAKE_SYSTEM_NAME MATCHES "Linux")
+-  find_package(PkgConfig REQUIRED)
+-  pkg_check_modules(UUID REQUIRED uuid)
++  find_package(libuuid REQUIRED)
+ endif()
+ if(APPLE)
+   find_library(COREFOUNDATION_LIBRARY CoreFoundation)
+--- runtime/CMakeLists.txt
++++ runtime/CMakeLists.txt
+@@ -76,8 +76,8 @@
+ endif()
+ 
+ if(CMAKE_SYSTEM_NAME MATCHES "Linux")
+-  target_link_libraries(antlr4_shared ${UUID_LIBRARIES})
+-  target_link_libraries(antlr4_static ${UUID_LIBRARIES})
++  target_link_libraries(antlr4_shared ${libuuid_LIBRARIES})
++  target_link_libraries(antlr4_static ${libuuid_LIBRARIES})
+ elseif(APPLE)
+   target_link_libraries(antlr4_shared ${COREFOUNDATION_LIBRARY})
+   target_link_libraries(antlr4_static ${COREFOUNDATION_LIBRARY})
```

Comment:
```suggestion
   target_link_libraries(antlr4_static ${COREFOUNDATION_LIBRARY})
@@ -141,7 +141,7 @@
                                  COMPILE_FLAGS "${disabled_compile_warnings} ${extra_static_compile_flags}")
 
 install(TARGETS antlr4_shared
-        DESTINATION lib
-        EXPORT antlr4-targets)
+        EXPORT antlr4-targets
+        ARCHIVE DESTINATION lib LIBRARY DESTINATION lib RUNTIME DESTINATION bin)
 install(TARGETS antlr4_static
         DESTINATION lib
```

Reply:
Got:

~~~ 
ERROR: antlr4-cppruntime/4.9.3: Error in build() method, line 74
	self._patch_sources()
while calling '_patch_sources', line 41
	tools.patch(**patch)
	ConanException: Failed to apply patch: patches/0001-update-cmakelist.patch
~~~

Did you test it ?

Reply:
This Patchfile is malformed.

Reply:
Additional spaces in 27

Reply:
Please wait with applying my suggestions.
I'll open a pr.

Reply:
I've already applied and testing.
Fixed patch so far.
But rest also are failing.

Reply:
~~~
--- runtime/Cpp/CMakeLists.txt
+++ runtime/Cpp/CMakeLists.txt
@@ -40,8 +40,7 @@
 endif()
 
 if(CMAKE_SYSTEM_NAME MATCHES "Linux")
-  find_package(PkgConfig REQUIRED)
-  pkg_check_modules(UUID REQUIRED uuid)
+  find_package(libuuid REQUIRED)
 endif()
 if(APPLE)
   find_library(COREFOUNDATION_LIBRARY CoreFoundation)
diff -ruN /home/mwidera/.conan/data/antlr4-cppruntime/4.9.3/_/_/source/source_subfolder/runtime/Cpp/runtime/CMakeLists.txt mod/source_subfolder/runtime/Cpp/runtime/CMakeLists.txt
--- runtime/Cpp/runtime/CMakeLists.txt
+++ runtime/Cpp/runtime/CMakeLists.txt
@@ -76,8 +76,8 @@
 endif()
 
 if(CMAKE_SYSTEM_NAME MATCHES "Linux")
-  target_link_libraries(antlr4_shared ${UUID_LIBRARIES})
-  target_link_libraries(antlr4_static ${UUID_LIBRARIES})
+  target_link_libraries(antlr4_shared ${libuuid_LIBRARIES})
+  target_link_libraries(antlr4_static ${libuuid_LIBRARIES})
 elseif(APPLE)
   target_link_libraries(antlr4_shared ${COREFOUNDATION_LIBRARY})
   target_link_libraries(antlr4_static ${COREFOUNDATION_LIBRARY})
@@ -141,8 +141,8 @@
                                  COMPILE_FLAGS "${disabled_compile_warnings} ${extra_static_compile_flags}")
 
 install(TARGETS antlr4_shared
-        DESTINATION lib
-        EXPORT antlr4-targets)
+        EXPORT antlr4-targets
+        ARCHIVE DESTINATION lib LIBRARY DESTINATION lib RUNTIME DESTINATION bin)
 install(TARGETS antlr4_static
         DESTINATION lib
         EXPORT antlr4-targets)
~~~

Reply:
Next build failure:

~~~
RROR: antlr4-cppruntime/4.9.3: Error in build() method, line 75
	cmake = self._configure_cmake()
while calling '_configure_cmake', line 70
	self._cmake.configure(build_folder=self._build_subfolder, source_folder=self._source_subfolder)
	ConanException: Error 1 while executing cd '/home/mwidera/.conan/data/antlr4-cppruntime/4.9.3/_/_/build/ed4c50cce248f963b530d644f4c27bd3c33aa7ed/build_subfolder' && cmake -G "Unix Makefiles" -DCMAKE_BUILD_TYPE="Release" -DCONAN_IN_LOCAL_CACHE="ON" -DCONAN_COMPILER="gcc" -DCONAN_COMPILER_VERSION="9" -DCONAN_CXX_FLAGS="-m64" -DCONAN_SHARED_LINKER_FLAGS="-m64" -DCONAN_C_FLAGS="-m64" -DCONAN_LIBCXX="libstdc++11" -DBUILD_SHARED_LIBS="OFF" -DCMAKE_INSTALL_PREFIX="/home/mwidera/.conan/data/antlr4-cppruntime/4.9.3/_/_/package/ed4c50cce248f963b530d644f4c27bd3c33aa7ed" -DCMAKE_INSTALL_BINDIR="bin" -DCMAKE_INSTALL_SBINDIR="bin" -DCMAKE_INSTALL_LIBEXECDIR="bin" -DCMAKE_INSTALL_LIBDIR="lib" -DCMAKE_INSTALL_INCLUDEDIR="include" -DCMAKE_INSTALL_OLDINCLUDEDIR="include" -DCMAKE_INSTALL_DATAROOTDIR="share" -DCONAN_CMAKE_POSITION_INDEPENDENT_CODE="ON" -DCMAKE_MODULE_PATH="/home/mwidera/.conan/data/antlr4-cppruntime/4.9.3/_/_/build/ed4c50cce248f963b530d644f4c27bd3c33aa7ed" -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY="ON" -DCONAN_EXPORTED="1" -Wno-dev '/home/mwidera/.conan/data/antlr4-cppruntime/4.9.3/_/_/build/ed4c50cce248f963b530d644f4c27bd3c33aa7ed/source_subfolder'
~~~

Reply:
I said that this
~~~
extracted_dir = os.path.join( self._main_code_subfolder , "runtime", "Cpp" )
~~~
Is required

Reply:
If we revert this os.path.join - we need to modify Patchfile as well.
As I understand
~~~
cmake = self._configure_cmake()
~~~
Cannot work in subfolders - or I do not know how to make this working.
Therefore 
~~~
os.rename(extracted_dir, self._source_subfolder)
~~~ 
is called.
Without this you will not build this library.

Reply:
ok - your solution works.